api/test/integration/import-data/xlsx-uploads/import-data-produce-jobs.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { Queue } from 'bull';
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppModule } from 'app.module';
+import { ImportDataModule } from 'modules/import-data/import-data.module';
+import { getQueueToken } from '@nestjs/bull';
+import { importQueueName } from 'modules/import-data/workers/import-data.producer';
+import { ImportDataService } from 'modules/import-data/import-data.service';
+
+describe('XLSX Upload Feature Job Producer Tests', () => {
+  const importQueue: any = { add: jest.fn() };
+  const importQueueFail: any = { add: jest.fn().mockRejectedValue('test') };
+  let queue: Queue;
+  let moduleFixture: TestingModule;
+  let importDataService: ImportDataService;
+
+  const bootstrapTestingApp = async (fakeQueue: any): Promise<void> => {
+    moduleFixture = await Test.createTestingModule({
+      imports: [AppModule, ImportDataModule],
+    })
+      .overrideProvider(getQueueToken(importQueueName.value))
+      .useValue(fakeQueue)
+      .compile();
+
+    queue = moduleFixture.get(getQueueToken(importQueueName.value));
+    importDataService = moduleFixture.get(ImportDataService);
+  };
+
+  afterEach(async () => {
+    jest.clearAllMocks();
+    await moduleFixture.close();
+  });
+
+  test('When loadXlsxFile is called with required file data and a userId, Then a job should be pushed to the queue', async () => {
+    await bootstrapTestingApp(importQueue);
+    const fileData = { path: 'filepath' } as Express.Multer.File;
+    const userId = '29b47625-9ffa-45a9-bfce-7a75abeb820e';
+    await importDataService.loadXlsxFile(userId, fileData);
+
+    expect(queue.add).toHaveBeenCalledWith('excel-import-job', {
+      xlsxFilePath: fileData.path,
+      userId,
+    });
+  });
+  test('When loadXlsxFile is called with required file data and a userId, but the Job can not be added to the queue, a error should be displayed', async () => {
+    await bootstrapTestingApp(importQueueFail);
+    const fileData = {
+      path: 'filepath',
+      filename: 'filename',
+    } as Express.Multer.File;
+    const userId = '29b47625-9ffa-45a9-bfce-7a75abeb820e';
+    expect.assertions(1);
+    try {
```

Comment:
did you try something like https://stackoverflow.com/questions/46042613/how-to-test-the-type-of-a-thrown-exception-in-jest instead? seems cleaner and more intentional

Reply:
I honestly prefer my current approach as it lets you do more stuff with exceptions without relying on the jest API
Also I just gave it a try and for some reasons the expect block does not catch the exception, tried with .toThrow, toTrowError and a couple of methods more.

So taking in account that the test is accurate and for the sake of pragmatism, I'd suggest to leave it as is, because the flow will most likely change. We can add this to lightweight tech debt if you are OK with that